src/node/ext/ldap/_node.py

```diff
@@ -304,11 +306,18 @@ def __iter__(self):
                 res = list()
             if isinstance(res, tuple):
                 res, cookie = res
-            for dn, _ in res:
-                key = decode(explode_dn(dn)[0])
-                # do not yield if node is supposed to be deleted
-                if key not in self._deleted_children:
-                    yield key
+            if self.root._use_unicode:
+                for dn, _ in res:
+                    key = decode(explode_dn(dn)[0])
+                    # do not yield if node is supposed to be deleted
+                    if key not in self._deleted_children:
+                        yield key
+            else:
+                for dn, _ in res:
+                    key = explode_dn(dn)[0]
+                    # do not yield if node is supposed to be deleted
+                    if key not in self._deleted_children:
+                        yield key
```

Comment:
here copy pasting the whole logic section is not ok. When passing the current node to ``decode`` this is not needed any more. This may need some millisecs more, but I doubt it has a real impact.